cartography/intel/aws/iam.py

```diff
@@ -227,6 +227,16 @@ def get_account_access_key_data(boto3_session: boto3.session.Session, username:
         logger.warning(
             f"Could not get access key for user {username} due to NoSuchEntityException; skipping.",
         )
+    # We add Last Used data to each access key
+    access_keys_metadata = access_keys.get('AccessKeyMetadata', [])
+    for access_key in access_keys_metadata:
+        access_key_id = access_key.get('AccessKeyId')
+        if not access_key_id:
+            access_key['LastUsed'] = None
+            continue
+        last_used_result = client.get_access_key_last_used(AccessKeyId=access_key_id)
+        last_used_info = last_used_result.get('AccessKeyLastUsed', {"LastUsedDate": None})
+        access_key['LastUsed'] = last_used_info.get('LastUsedDate', None)
```

Comment:
Some of these fields are required, according to docs. Did you find that `AccessKeyMetadata` or `AccessKeyId` weren't always there?

```suggestion
    access_keys_metadata = access_keys['AccessKeyMetadata']
    for access_key in access_keys_metadata:
        access_key_id = access_key['AccessKeyId']
        last_used_result = client.get_access_key_last_used(AccessKeyId=access_key_id)
        last_used_info = last_used_result.get('AccessKeyLastUsed', {"LastUsedDate": None})
        access_key['LastUsed'] = last_used_info.get('LastUsedDate', None)
```

Reply:
i've been bitten by APIs a lot in the past -- but -- point taken